server/task_executor_pitr_restore.go

```diff
@@ -38,6 +39,83 @@ func (exec *PITRRestoreTaskExecutor) RunOnce(ctx context.Context, server *Server
 		return true, nil, fmt.Errorf("invalid PITR restore payload: %s, error: %w", task.Payload, err)
 	}
 
+	if (payload.BackupID == nil) == (payload.PointInTimeTs == nil) {
+		return true, nil, fmt.Errorf("only one of BackupID and time point can be set")
+	}
+
+	if payload.BackupID != nil {
+		// Restore full backup only
+		backup, err := server.store.GetBackupByID(ctx, *payload.BackupID)
+		if err != nil {
+			return true, nil, fmt.Errorf("failed to find backup with ID %d, error: %w", *payload.BackupID, err)
+		}
+		if backup == nil {
+			return true, nil, fmt.Errorf("backup with ID %d not found", *payload.BackupID)
+		}
+		sourceDatabase, err := server.store.GetDatabase(ctx, &api.DatabaseFind{ID: &backup.DatabaseID})
+		if err != nil {
+			return true, nil, fmt.Errorf("failed to find database for the backup: %w", err)
+		}
+		if sourceDatabase == nil {
+			return true, nil, fmt.Errorf("source database ID not found %v", backup.DatabaseID)
+		}
+		targetDatabaseFind := &api.DatabaseFind{
+			InstanceID: &task.InstanceID,
```

Comment:
If payload.TargetInstanceID != nil, we should use that.

Reply:
For now, we just restore the full backup to a new database in the same instance. However, for generality, it can be modified like this. SGTM : )